pkg/kubelet/kubelet.go

```diff
@@ -1299,6 +1301,14 @@ func (kl *Kubelet) syncPod(pod *api.BoundPod, containersInPod dockertools.Docker
 		}
 	}
 
+	// Regenerate status and store it in the cache.
```

Comment:
I think we may want to do this in a defer, if we error out on any part of the function we won't generate the status.